hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -594,14 +610,24 @@ private void handleFlushInternal(boolean close)
     if (totalDataFlushedLength < writtenDataLength) {
       refreshCurrentBuffer();
       Preconditions.checkArgument(currentBuffer.position() > 0);
-      if (currentBuffer.hasRemaining()) {
-        writeChunk(currentBuffer);
-      }
+
       // This can be a partially filled chunk. Since we are flushing the buffer
       // here, we just limit this buffer to the current position. So that next
       // write will happen in new buffer
-      updateFlushLength();
-      executePutBlock(close, false);
+      if (currentBuffer.hasRemaining()) {
+        if (writtenDataLength - totalDataFlushedLength < SMALL_CHUNK_THRESHOLD &&
```

Comment:
@jojochuang , I think we can always use write chunk + put block if currentBuffer.hasRemaining is true.

Reply:
Agreed. Will update the PR.